homeassistant/components/media_player/webostv.py

```diff
@@ -56,32 +57,37 @@
     vol.Optional(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
-    vol.Optional(CONF_TIMEOUT, default=8): cv.positive_int,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
 })
 
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the LG WebOS TV platform."""
     if discovery_info is not None:
-        host = urlparse(discovery_info[1]).hostname
+        host = discovery_info.get('host')
+        name = discovery_info.get('name', DEFAULT_NAME)
+        customize = {}
+        timeout = DEFAULT_TIMEOUT
+        turn_on_action = None
+        config_file_path = WEBOSTV_CONFIG_FILE
     else:
         host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        customize = config.get(CONF_CUSTOMIZE)
+        timeout = config.get(CONF_TIMEOUT)
+        turn_on_action = config.get(CONF_ON_ACTION)
+        config_file_path = config.get(CONF_FILENAME)
 
-    if host is None:
-        _LOGGER.error("No TV found in configuration file or with discovery")
-        return False
+        if host is None:
+            _LOGGER.error("No TV found in configuration file")
```

Comment:
I've only update the text, however this should be fixed with a different PR because that text was a lie for several reasons:
- PLATFORM_SCHEMA and also the docs say that the host is optional which from the code we can see that it will complain that it's missing
- specifying that no tv was found is another lie, because autodiscovery runs after manual configuration and never got a chance to specify the host discovered by netdisco
- with the current code users which don't specify a host will see that error in their logs, however after the tv is discovered it will be configured correctly and the users can use it normally

PS: I couldn't fix this myself since it may require changes also in the pylgtv package that is maintained by @TheRealLink and who also wrote the initial version of this HA component but at least now both autodiscovery and of course manual configuration (with host specified) work

Reply:
You can set it as required